### PR TITLE
[bitnami/flink] Add VIB tests

### PR DESCRIPTION
**Pull request by andresbono**
### Description of the change

The main objective of this PR is to publish our Bitnami Flink container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

### CODE REVIEW

**Comment by andresbono** [comment]
Pending on new release of `bitnami/flink` fixing some minor issues found

**Comment by andresbono** [comment]
Tests passing: https://github.com/bitnami/containers/actions/runs/5054640771/jobs/9069806850

File: .vib/flink/goss/vars.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+root_dir: /opt/bitnami
+version:
+  bin_name: flink
+  flag: -v
+user:
+  flink:
+    exists: true
+group:
+  flink:
+    exists: true
```
Comment (migruiz4) [inline .vib/flink/goss/vars.yaml:20]: User and group tests are not rendered in the `vars.yaml`.

Instead, you should move them under `flink.yaml`

File: .vib/flink/goss/flink.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    stdout:
+      - Rest endpoint listening
+      - Web frontend listening
+file:
+  /opt/bitnami/flink/plugins/s3-fs-base/flink-s3-fs-base-{{ .Env.APP_VERSION }}.jar:
+    exists: true
+  /opt/bitnami/flink/plugins/s3-fs-presto/flink-s3-fs-presto-{{ .Env.APP_VERSION }}.jar:
+    exists: true
+  /opt/bitnami/flink/plugins/s3-fs-hadoop/flink-s3-fs-hadoop-{{ .Env.APP_VERSION }}.jar:
+    exists: true
```
Comment (migruiz4) [inline .vib/flink/goss/flink.yaml:15]: I would suggest adding the `filetype` entry here:
```suggestion
  /opt/bitnami/flink/plugins/s3-fs-base/flink-s3-fs-base-{{ .Env.APP_VERSION }}.jar:
    exists: true
    filetype: file
  /opt/bitnami/flink/plugins/s3-fs-presto/flink-s3-fs-presto-{{ .Env.APP_VERSION }}.jar:
    exists: true
    filetype: file
  /opt/bitnami/flink/plugins/s3-fs-hadoop/flink-s3-fs-hadoop-{{ .Env.APP_VERSION }}.jar:
    exists: true
    filetype: file
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Please take a look at my comments

File: .vib/flink/goss/vars.yaml (hunk 1 above)
Comment (andresbono) [reply .vib/flink/goss/vars.yaml:20]: You are right! Good catch!

**Comment by andresbono** [comment]
Tests passing: https://github.com/bitnami/containers/actions/runs/5110032596/jobs/9185471277

**Review by migruiz4 — APPROVED** [review]
Looks good to me!

**Pull request merged**